### ISSUE TITLE
Sometimes, at almost complete XDCC downloads, I get "Connection reset by peer" -- resuming never completes (but, resuming that same file with AdiIRC it completes ok)

### ISSUE THREAD

**Issue by darkred** [kvirc/KVIrc#2270]
Using KVIrc, 4.9.2 (latest Git build, 2017-08-25) in win10x64.

So far the following issues has occurred two times:
while downloading (.tar files) from XDCC bots.
as the download reaches 100% (but hasn't fully completed), I get "Connection reset by peer".
Then, no matter how many times I try resuming it, 
it downloads some kB, and then it's almost always "Transfer failed: Connection reset by peer" (or "Transfer failed: Unknown error").
Then if I try to open the (uncompleted, .tar) file in WinRAR or 7-zip 
I get "Unexpected end of data" for the archive, and "Data Error" for various contained rar parts.

On the other hand, 
if I copy that uncompleted .tar file
and resume it with AdiIRC 2.9 (latest beta) it completes ok immediately,
and the resulting archive has no problems in WinRAR/7-zip.

Here is the log from the Transfers tab: 
(*I've removed the remote addresses - I'll post the address for that XDCC download in the kvirc IRC channel , if anyone wants to try it*)
```
[14:09:03] [TRANSFER 5]: Transferring data
[16:20:40] [TRANSFER 5]: Transfer failed: Connection reset by peer
[16:49:32] [TRANSFER 6]: Attempting an active DCC RECV connection
[16:49:32] [TRANSFER 6]: Sent DCC RESUME request to ___, waiting for ACCEPT
[16:49:34] [TRANSFER 6]: RESUME accepted, transfer will begin at position 35617710232
[16:49:35] [TRANSFER 6]: Contacting host ___ on port 31000
[16:49:35] [TRANSFER 6]: Connected to ___:31000
[16:49:35] [TRANSFER 6]: Local end is 192.168.1.66:51771
[16:49:35] [TRANSFER 6]: Transferring data
[16:49:36] [TRANSFER 6]: Transfer failed: Connection reset by peer
[16:50:01] [TRANSFER 7]: Attempting an active DCC RECV connection
[16:50:01] [TRANSFER 7]: Sent DCC RESUME request to ___, waiting for ACCEPT
[16:50:03] [TRANSFER 7]: RESUME accepted, transfer will begin at position 35618177632
[16:50:03] [TRANSFER 7]: Contacting host ___ on port 31000
[16:50:03] [TRANSFER 7]: Connected to ___:31000
[16:50:03] [TRANSFER 7]: Local end is 192.168.1.66:51772
[16:50:03] [TRANSFER 7]: Transferring data
[16:50:04] [TRANSFER 7]: Transfer failed: Connection reset by peer
[17:00:00] [TRANSFER 8]: Attempting an active DCC RECV connection
[17:00:00] [TRANSFER 8]: Sent DCC RESUME request to ___, waiting for ACCEPT
[17:00:08] [TRANSFER 8]: RESUME accepted, transfer will begin at position 35618263764
[17:00:08] [TRANSFER 8]: Contacting host ___ on port 31000
[17:00:08] [TRANSFER 8]: Connected to ___:31000
[17:00:08] [TRANSFER 8]: Local end is 192.168.1.66:52190
[17:00:08] [TRANSFER 8]: Transferring data
[17:00:08] [TRANSFER 8]: Transfer failed: Unknown error
```

### System information
[//]:# (Type ```/about.kvirc``` or click "Help → About KVIrc" and copy & paste the details from the "Executable Information" tab below between ``` marks)  
[//]:# (Also Enter any OS information the "Executable Information" tab doesn't contain, e.g. KDE, Gentoo, Linux Mint, etc.)

```
KVIrc 4.9.2 'Aria'

Runtime Info:
System name: Windows Release : 10.0.15063
System version: Windows 10 Professional (x64) (Build 15063)
Architecture: Intel64 Family 6 Model 60 Stepping 3
Qt version: 5.8.0
Qt theme: windowsvista

Build Info:
Build date: 2017-08-25 12:46:35 UTC
Sources date: 20170825
Revision number: git-7150-gcd3bf0ef5
System name: Windows-6.3.9600
CPU name: AMD64
Build command: C:/Program Files (x86)/CMake/bin/cmake.exe
Build flags: 
   MANDIR=share/man
   CMAKE_INSTALL_PREFIX=C:/projects/kvirc/build/release
   Threads=Win32
Compiler name: C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe
Compiler flags: /DWIN32 /D_WINDOWS /W3 /GR /EHsc
Qt version: 5.8.0
Features: IRC, IPv6, Crypt, SSL, IPC, Transparency, Perl, Python, Enchant, Qt5, KVS
```

**Comment by ElonSatoshi** [comment]
This is not a KVIrc issue, this is an internet issue. It happens universally. It happens in Weechat. It happens in Hexchat. It happens in Firefox. It happens in wget.

I don't even use KVIrc, I happened to be in #kvirc and I saw this issue pop up.

You'll have to get a better internet and/or try to resume the download with KVIrc. If you have a nasty ISP that censors things with connection reset, try tor or a vpn.

**Comment by darkred** [comment]
That's definitely not the case... 

I didn't have an internet disconnect at that time, and I don't have any problem with my ISP.

The "Connection reset by peer" occurred while only 0.6 MB was remaining for the download to complete,
i.e. it was showing already at 100% in the Progress column.

What I did was this:
at the first error, I copied the unfinished file to another folder,
then I tried resuming the current file 3 times with KVIrc, always the same error.
Then I launched AdiIRC and tried resuming the **copied** file: it immediately completed.
